Graphics/GraphicsEngineD3D11/src/ShaderD3D11Impl.cpp

```diff
@@ -34,24 +34,24 @@
 namespace Diligent
 {
 
-static const std::string HLSLVersionToShaderModelString(const ShaderVersion& Version, Uint8 MaxMajorRevision, Uint8 MaxMinorRevision)
+static const ShaderVersion HLSLValidateShaderVersion(const ShaderVersion& Version, const ShaderVersion& MaxVersion)
 {
-    std::string ModelStr;
-    if (Version.Major > MaxMajorRevision || Version.Major == MaxMajorRevision && Version.Minor > MaxMinorRevision)
+    ShaderVersion ModelVer;
+    if (Version.Major > MaxVersion.Major || (Version.Major == MaxVersion.Major && Version.Minor > MaxVersion.Minor))
     {
-        ModelStr = std::to_string(Uint32{MaxMajorRevision}) + '_' + std::to_string(Uint32{MaxMinorRevision});
+        ModelVer = MaxVersion;
         LOG_ERROR_MESSAGE("Shader model ", Uint32{Version.Major}, "_", Uint32{Version.Minor},
                           " is not supported by this device. Maximum supported model: ",
-                          ModelStr, ". Attempting to use ", ModelStr, '.');
+                          MaxVersion.Major, "_", MaxVersion.Minor, ". Attempting to use ", MaxVersion.Major, "_", MaxVersion.Minor, '.');
```

Comment:
Note that these are Unit8, which so is should be Uint32{MaxVersion.Major}. Fixed it in my branch.